.github/workflows/build-and-test.yaml

```diff
@@ -1,6 +1,10 @@
 name: build-and-test-scheduled
 
 on:
+  push:
```

Comment:
`build-and-test-scheduled` takes about 1~2 hours.
What happens if new PRs are submitted before `build-and-test-scheduled` completes?

Reply:
In this case, CodeCov cannot find base commit. 
CodeCov will display the following warnings in its comment: https://github.com/autowarefoundation/autoware.universe/pull/292#issuecomment-1020797172

Reply:
Then, I feel there is no reason for adding push events. :thinking: 
Do we really need it? Could you explain the reason again, please?

Reply:
You are probably concerned that adding a push event will not get the correct Diff.
In my opinion, only `build-and-test-scheduled` leaves incomplete coverage records, and CodeCov cannot create Diffs. That's why we need the push event.

Also, I found a new document that says:

Codecov checks all your CI statuses on **every commit** to make sure the coverage results are accurate. This process is very important to the inner working of Codecov and data accuracy.

https://docs.codecov.com/docs/ci-service-relationship#if-ci-is-pending-then-the-coverage-results

Therefore, we really need it.

Reply:
I see, but since we're using self-hosted runner for this, I don't want to run this workflow for every commit.
How about preparing a new workflow such as `update-coverage.yaml`?

Reply:
Or splitting the ARM build job from `build-and-test-scheduled.yaml` is okay.

Reply:
Is running the coverage job on x64 machine acceptable to you?
I don't think CPU architecture matters much for coverage, so I think it is enough to run it on x64 machine.

Reply:
Yes, it's fine.

Reply:
Split a workflow file.
b5a1da31